### PR TITLE
 Adding crossgen build

### PR DESCRIPTION
**Pull request by alrod**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -22,26 +19,151 @@ function ZipContent([string] $sourceDirectory, [string] $target)
...
 }
 
+function CrossGen([string] $runtime, [bool] $isSelfContained, [string] $publishTarget, [string] $privateSiteExtensionPath)
+{
+    Write-Host "publishTarget: " $publishTarget
+    Write-Host "privateSiteExtensionPath: " $privateSiteExtensionPath
+
+    $selfContained = Join-Path $publishTarget "self-contained"
+    $regular = Join-Path $publishTarget "regular"
+    $croosGen = "$publishTarget\download\crossgen\crossgen.exe"
```
Comment (brettsam) [inline build.ps1]: nit: spelling of this variable

File: build.ps1 [hunk 2]
```diff
@@ -22,26 +19,151 @@ function ZipContent([string] $sourceDirectory, [string] $target)
...
+    DownloadNupkg "https://www.nuget.org/api/v2/package/Microsoft.Build.Tasks.Core/15.1.1012" @("lib\netstandard1.3\Microsoft.Build.Tasks.Core.dll")  @("$selfContained")
+    DownloadNupkg "https://www.nuget.org/api/v2/package/Microsoft.Build.Utilities.Core/15.1.1012" @("lib\netstandard1.3\Microsoft.Build.Utilities.Core.dll")  @("$selfContained")
+
+    # Publish self-contained app with all required dlls for crossgen
+    dotnet publish .\src\WebJobs.Script.WebHost\WebJobs.Script.WebHost.csproj -r $runtime -o "$selfContained" -v q /p:BuildNumber=$buildNumber    
+
+    # For self contained we want to process only IL ddls
+    if ($isSelfContained) {
+        dotnet publish .\src\WebJobs.Script.WebHost\WebJobs.Script.WebHost.csproj -o $regular -v q /p:BuildNumber=$buildNumber
+    }
```
Comment (brettsam) [inline build.ps1]: This looks confusing -- you check "if selfcontained", but you're publishing to "regular", yet you've already published to "selfContained". Is a variable name wrong somewhere?

File: build.ps1 [hunk 3]
```diff
@@ -22,26 +19,151 @@ function ZipContent([string] $sourceDirectory, [string] $target)
   [IO.Compression.ZipFile]::CreateFromDirectory($sourceDirectory, $target)
 }
 
+function CrossGen([string] $runtime, [bool] $isSelfContained, [string] $publishTarget, [string] $privateSiteExtensionPath)
+{
+    Write-Host "publishTarget: " $publishTarget
+    Write-Host "privateSiteExtensionPath: " $privateSiteExtensionPath
+
+    $selfContained = Join-Path $publishTarget "self-contained"
+    $regular = Join-Path $publishTarget "regular"
```
Comment (brettsam) [inline build.ps1]: Can you explain what "self-contained" vs "regular" means?

Comment (alrod) [reply build.ps1]: Self-contained deployment contains all NET Core and ASP NET Core resources required for the application.
More info: https://docs.microsoft.com/en-us/dotnet/core/deploying/

File: build.ps1 (hunk 2 above)
Comment (alrod) [reply build.ps1]: we publish to "regular" as we want to crossgen only our custom dlls. 
So we have two folders: "Self-Contained" and "Regular".
We will be processing all dlls which are in "self-contained" folder and in "Regular" folder with output to "self-contained".

**Pull request merged**